Jenkinsfile-nightly

```diff
@@ -0,0 +1,78 @@
+            // when { branch 'master' } 
+            steps {
+                script{
+                    for(i in OLTP_TERMINALS){
+                        sh 'echo $i'
+                    }
+                }
+                sh 'echo $NODE_NAME'
+                sh 'echo y | sudo ./script/installation/packages.sh all'
+                sh 'mkdir build'
+                // We run tatp/smallbank/noop/ycsb with the default config (--scale-factor=1 --terminals=1)
+                sh 'cd build && cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Release -DTERRIER_USE_ASAN=OFF -DTERRIER_USE_JEMALLOC=ON -DTERRIER_BUILD_TESTS=OFF .. && make -j$(nproc) all'
+                // sh 'cd build && python3 ../script/testing/oltpbench/run_oltpbench.py tatp 2,35,10,35,2,14,2 --build-type=release --loader-threads=4'
+                // sh 'cd build && python3 ../script/testing/oltpbench/run_oltpbench.py smallbank 15,15,15,25,15,15 --build-type=release --loader-threads=4'
+                // sh 'cd build && python3 ../script/testing/oltpbench/run_oltpbench.py noop 100 --build-type=release'
+                // sh 'cd build && python3 ../script/testing/oltpbench/run_oltpbench.py ycsb 50,5,15,10,10,10 --build-type=debug --loader-threads=4'
+
+                // For TPC-C, we use scalefactor 4, and scale the number of terminals (1,2,4,8,16)
+                loop_terminals_run_oltp(terminals)
+                archiveArtifacts 'build/oltpsummary/*.json'
+            }
+            post {
+                cleanup {
+                    deleteDir()
+                }
+            }
+        }
+
+        stage('Microbenchmark') {
+            agent { label 'benchmark' }
+            // TODO remove when merge
+            // when { branch 'master' }
+            steps {
+                sh 'echo $NODE_NAME'
+                sh 'echo y | sudo ./script/installation/packages.sh all'
+                sh 'mkdir build'
+                sh 'cd build && cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=Release -DTERRIER_USE_ASAN=OFF -DTERRIER_USE_JEMALLOC=ON -DTERRIER_BUILD_TESTS=OFF .. && make -j$(nproc) all'
+                // The micro_bench configuration has to be consistent because we currently check against previous runs with the same config
+                //  # of Threads: 4
+                //  WAL Path: Ramdisk
+                sh 'cd script/micro_bench ./run_micro_bench.py --run --num-threads=4 --logfile-path=/mnt/ramdisk/benchmark.log'
+                archiveArtifacts 'script/micro_bench/*.json'
+                junit 'script/micro_bench/*.xml'
+            }
+            post {
+                cleanup {
+                    deleteDir()
+                }
+            }
+        }
+    }
+    
+}
+
+@NonCPS
+def loop_terminals_run_oltp(list){
+    list.each { item ->
+        sh 'cd build && python3 ../script/testing/oltpbench/run_oltpbench.py tpcc 45,43,4,4,4 --build-type=release --loader-threads=4 --scale-factor=4 --terminals=${item}'
```

Comment:
This should include all the benchmark sets

Reply:
It is still in progress